File: configlibrary/src/main/java/com/shmedo/configlibrary/iot/parser/DeviceCurrentStateParser.java
package com.shmedo.configlibrary.iot.parser;

import com.shmedo.configlibrary.iot.enums.IOTCommandType;
import com.shmedo.configlibrary.iot.interfaces.IOTResultParser;

/**
 * 创建者:   gonghe <br/>
 * 创建时间:  2020/8/31 <br/>
 * 描述：     TODO
 */
public class DeviceCurrentStateParser implements IOTResultParser<String> {
    @Override
    public String parse(String result) {
        String[] strs = result.split("&");
        return strs[1];
    }

    @Override
    public void validate(String result) {

    }

    @Override
    public IOTCommandType commandType() {
        return IOTCommandType.QUERY_DEVICE_STATUS;
    }
}

File: lib_device_base/src/main/java/com/shmedo/lib/device/base/iot_cmd/parser/DeviceCurrentStateParser.kt
package com.shmedo.lib.device.base.iot_cmd.parser

import com.shmedo.lib.device.base.iot_cmd.interfaces.IOTCommandResponseParser
import com.shmedo.lib.device.base.iot_cmd.enums.IOTCommandType

/**
 * 创建者:   gonghe <br/>
 * 创建时间:  2023/9/22 <br/>
 * 描述：     TODO
 */
class DeviceCurrentStateParser: IOTCommandResponseParser<String> {
    override fun parseInstance(keyValueMap: Map<String, String>): String {
        return keyValueMap["state"]!!
    }

    override fun commandType(): IOTCommandType = IOTCommandType.QUERY_DEVICE_STATUS
}